### PR TITLE
Sensor Monitor support

### PR DESCRIPTION
**Pull request by bmridul**
Sonic Platform common changes required as part of Voltage & Current Sensor Monitoring Feature.
#### Description

New base class is introduced for defining voltage and current sensors.
Chassis and Module base classes are enhanced with APIs to retrieve voltage and current sensor data from platform.
#### Motivation and Context

The changes are required to support the design for sensor monitoring feature as proposed in the HLD.

HLD : https://github.com/sonic-net/SONiC/pull/1394
#### How Has This Been Tested?

Tested with CLIs for sensor monitoring.

### CODE REVIEW

File: sonic_platform_base/module_base.py [hunk 1]
```diff
@@ -372,6 +374,95 @@ def get_thermal(self, index):
...
+            sys.stderr.write("Voltage sensor index {} out of range (0-{})\n".format(
+                             index, len(self._voltage_sensor_list)-1))
+
+        return voltage_sensor
+
+    ##############################################
+    # Current sensor methods
+    ##############################################
+
+    def get_num_CurrentSensors(self):
```
Comment (Junchao-Mellanox) [inline sonic_platform_base/module_base.py:426]: get_num_current_sensors?

File: sonic_platform_base/sensor_base.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+SENSOR_TYPE_CURRENT = 2
+
+class SensorBase(device_base.DeviceBase):
+    """
+    Abstract base class for interfacing with a sensor module
+    """
+
+    def get_type(self):
+        """
+        Specifies the type of the sensor.
```
Comment (Junchao-Mellanox) [inline sonic_platform_base/sensor_base.py:20]: What could be the valid type values?

File: sonic_platform_base/sensor_base.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+
+class VoltageSensorBase(SensorBase):
+    """
+    Abstract base class for interfacing with a voltage sensor module
+    """
+    MILLI_VOLTS = "mV"
+
+    def get_type(self):
+        return SENSOR_TYPE_VOLTAGE
```
Comment (Junchao-Mellanox) [inline sonic_platform_base/sensor_base.py:156]: should it be a string instead of integer?

File: sonic_platform_base/sensor_base.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+    def get_unit(self):
+        return MILLI_VOLTS
```
Comment (Junchao-Mellanox) [inline sonic_platform_base/sensor_base.py:159]: self.MILLI_VOLTS?

File: sonic_platform_base/sensor_base.py [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+    """
+    Abstract base class for interfacing with a current sensor module
+    """
+    MILLI_AMPS = "mA"
+
+    def get_type(self):
+        return SENSOR_TYPE_CURRENT
+
+    def get_unit(self):
+        return MILLI_AMPS
```
Comment (Junchao-Mellanox) [inline sonic_platform_base/sensor_base.py:172]: self.MILLI_AMPS?

**Review by Junchao-Mellanox — CHANGES_REQUESTED** [review]

**Review by Junchao-Mellanox — APPROVED** [review]

**Review by abdosi — APPROVED** [review]

**Comment by abdosi** [comment]
@gechiang this is enhancement so will need to go to msft repo. not sure if we can take into 202205 branch.

**Comment by gechiang** [comment]
> @gechiang this is enhancement so will need to go to msft repo. not sure if we can take into 202205 branch.

Unfortunately we currently do not have a separate MSFT Repo for sonic-platform-common so there is no way to take it in that way... either it is taken up in public 202205 or someone has to create the new MSFT repo for this submodule and also work with build team to enable auto sync works which was not that easy when we did it for sonic-utilities.msft  So unless it is absolutely needed, no plan to create a new msft repo for new features.

**Comment by StormLiangMS** [comment]
new feature, no cherry pick.

**Pull request merged**